### PR TITLE
Added SQL reconnection logic.

### LINKED ISSUE #514: A SQL error occured during this operation, check the server logs (lost playtime?)
Round 471

My character data was still there, though, and there were no times for the species.

Seems to correct itself once the round actually started

### PR DESCRIPTION
**Pull request by AndrewMontagne**
Adds some logic that automatically tries to reconnect to the MySQL server if the connection drops.

Stops me from being woken up at 4am by people telling me the database is down, fixes #514

### CODE REVIEW

File: code/controllers/subsystem/dbcore.dm [hunk 1]
```diff
@@ -188,11 +204,17 @@ Delayed insert mode was removed in mysql 7 and only works with MyISAM type table
...
-	if(!. && log_error)
-		log_sql("[ErrorMsg()] | Query used: [sql]")
+
+	if(!.)
+		var/error_msg = ErrorMsg()
+		if(log_error)
+			log_sql("[error_msg] | Query used: [sql]")
+		if(error_msg == "MySQL server has gone away" || error_msg == "Lost connection to MySQL server during query")
+			if(SSdbcore.AutoReconnect() && retries < 5)
+				return Execute(sql_query, cursor_handler, log_error, retries + 1)
```
Comment (AndrewMontagne) [inline code/controllers/subsystem/dbcore.dm]: image

File: code/controllers/subsystem/dbcore.dm [hunk 2]
```diff
@@ -69,6 +70,21 @@ SUBSYSTEM_DEF(dbcore)
 		log_sql("Connect() failed | [ErrorMsg()]")
 		++failed_connections
 
+/datum/controller/subsystem/dbcore/proc/AutoReconnect()
+	if(!auto_reconnecting)
+		auto_reconnecting = TRUE
+		message_admins("MySQL connection interrupted")
```
Comment (FlattestGuitar) [inline code/controllers/subsystem/dbcore.dm]: Add an `attempting to reconnect` here so admins don't freak out.

**Comment by ike709** [comment]
If this could be merged soon, that'd be great.

**Comment by FlattestGuitar** [comment]
Fuck it, we can add a better message later, this is High Priority.

**Pull request merged**